charts/consul/templates/_helpers.tpl

```diff
@@ -685,4 +681,17 @@ Usage: {{ template "consul.versionInfo" }}
     {{- $sanitizedVersion = $versionInfo }}
 {{- end -}}
 {{- printf "%s" $sanitizedVersion | quote }}
+{{- end -}}
+
+{{/*
+Validate that the depraced stanze for .Values.apiGateway is not set.
+Cleanup server.extraConfig entries to avoid conflicting entries:
+    - server.enableAgentDebug:
+      - `enable_debug` should not exist in extraConfig
+Usage: {{ template "consul.validateExtraConfig" . }}
```

Comment:
I don't think you need to use a helper template, we do an `if blah, fail` directly in a lot of the manifests already so you could just have:

```
{{- if .Values.apiGateway}}{{fail "[DEPRECATED and REMOVED]  apiGateway stanza is no longer supported as of Consul 1.19.0. Use connectInject.apiGateway instead."}}{{ end }}
```

at the top of the gateway-jobs yaml.

Also, I think say "apiGateway stanza" instead of ".Values.apiGateway" because users won't know what .Values is